### ISSUE TITLE
for loop with large int values returns negative numbers

### ISSUE THREAD

**Issue by dcuny** [godotengine/godot#83293]
### Godot version

Godot v4.1.1.stable

### System information

Godot v4.1.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated GeForce GT 640 () - AMD FX(tm)-8320 Eight-Core Processor (8 Threads)

### Issue description

`for` loops with large `int` values return negative values instead of the correct values.

### Steps to reproduce

Example code:

```
const BLICKS:int = 705600000
var start_loop:int = BLICKS * 40
var end_loop:int = BLICKS * 42
var step_loop: int = BLICKS / 32
for i in range( start_loop, end_loop, step_loop ):
		print(i)
```

Example output:
```

-1840771072
-1818721072
-1796671072
-1774621072
-1752571072
-1730521072
-1708471072
...
```

In contrast, a `while` loop returns the correct values:
```
const BLICKS:int = 705600000
var start_loop:int = BLICKS * 40
var end_loop:int = BLICKS * 42
var step_loop: int = BLICKS / 32
while start_loop < end_loop:
	print(start_loop)
	start_loop += step_loop
```
Correct output:
```
28224000000
28246050000
28268100000
28290150000
28312200000
28334250000
28356300000
28378350000
...
```

### Minimal reproduction project

N/A

**Comment by AThousandShips** [comment]
I think this is because internally the `range` argument uses a `Vector3i` which is 32 bits signed, CC @dalexeev

**Comment by AThousandShips** [comment]
Correction, it seems to be due to the fact that `range` internally counts with int, the vector based approach was before

**Comment by AThousandShips** [comment]
Can take a look at fixing this tomorrow

**Comment by dcuny** [comment]
Thanks for the prompt response!

**Comment by AThousandShips** [comment]
Got a basic solution working, but will have to look at some details, will open a PR tomorrow

**Comment by dalexeev** [comment]
> I think this is because internally the `range` argument uses a `Vector3i` which is 32 bits signed

Yes, the `for` loop it can iterate `Vector{2,2i,3,3i}`, the analyzer implicitly replaces the constant `range()` with an `int`, `float` or vector so as not to allocate an array unlike `range()` utility. There is also a PR #71564 for non-constant `range()`. I'm not sure there's an easy fix for this other than fixing/adding opcode to the GDScript VM.

**Comment by AThousandShips** [comment]
Replacing `int` with `int64_t` solves it at least in this specific case, will do some more testing tomorrow

**Comment by kleonc** [comment]
See #67025, in there I'm changing `range` to use `int64_t` but it also mentions the issue with the optimized version. Not sure how relevant/outdated that PR is though, it's kinda in limbo. If anyone wants to take it over / extract that part - feel free to do so.

**Comment by AThousandShips** [comment]
That is a more comprehensive fix indeed, will take a look at that too

**Comment by AThousandShips** [comment]
Might try work on this when I have more time, otherwise my above code and #67025 can be used as starting points for someone wanting to solve it
